### PR TITLE
Feature/rails upgrade

### PR DESCRIPTION
**Pull request by Obversity**
To bootstrap a project locally and test this branch outside of the dummy project, edit `lib/templates/application/app.rb` and comment out line 61 (`gem 'koi', github: 'katalyst/koi', branch: 'master'` and uncomment line 63 (`gem 'koi', path: File.join(File.dirname(__FILE__), '../../..')`
This is means when your project gets bootstrapped, it'll use your local Koi repo, not the one on github. 

I'd appreciate it if you'd try running it on your local machine to make sure the bootstrap process works and a new app will run off the bat.  Something like: 

```
cd ~/Projects
rbenv install 2.4.1
rbenv local 2.4.1
gem install bundler
gem install rails --version=5.1.0
rm -rf my_app && rails _5.1.0_ new my_app -d postgresql -m /Path/To/Your/Clone/koi/lib/templates/application/app.rb
```

Let me know if you've got any questions about any of the changes. 

Some of the new files were generated by the `rails app:update` command. 

TODO:
- when devise and simple_form have pushed rails 5.1 compatible versions to rubygems, re-enable in `koi.gemspec` and `lib/koi.rb`.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -13,11 +13,9 @@ gemspec
 # Koi
 gem 'koi_config'                , git: 'https://github.com/katalyst/koi_config.git'
 
-# i18n ActiveRecord backend
-gem 'i18n-active_record'        , git: 'https://github.com/svenfuchs/i18n-active_record.git',
-                                  branch: 'master'
-
-gem 'sass-rails', '~> 5.0.0'
-gem 'ornament', github: 'katalyst/ornament'
+gem 'ornament', github: 'katalyst/ornament', branch: 'feature/koi-3-rails-upgrade'
```
Comment (pvawser01) [inline Gemfile]: We will need to record somewhere to change this to master when ornament is released properly.

File: README.md [hunk 2]
```diff
@@ -5,10 +5,10 @@
...
 
 ```bash
-rbenv local 2.2.2
-# rvm use 2.2.2
-gem install rails -v 4.2.1
-rails _4.2.1_ new my_app -d postgresql -m https://raw.github.com/katalyst/koi/v2.2.1/lib/templates/application/app.rb
+rbenv local 2.4.1
+# rvm use 2.4.1
+gem install rails -v 5.1.0
+rails _5.1.0_ new my_app -d postgresql -m https://raw.github.com/katalyst/koi/v2.2.1/lib/templates/application/app.rb
```
Comment (pvawser01) [inline README.md]: Template path will need to change once we have a koi3 branch

File: db/migrate/20111228035029_create_nav_items.rb [hunk 3]
```diff
@@ -1,4 +1,4 @@
-class CreateNavItems < ActiveRecord::Migration
+class CreateNavItems < ActiveRecord::Migration[5.0]
```
Comment (pvawser01) [inline db/migrate/20111228035029_create_nav_items.rb]: Should these all be versioned to 5.1?

File: koi.gemspec [hunk 4]
```diff
@@ -18,23 +18,17 @@ Gem::Specification.new do |s|
...
-  s.add_dependency 'mime-types', '~> 2.3'
+  s.add_dependency 'mime-types'#, '~> 2.3'
 
-  s.add_dependency 'rails', '~> 4.2.1'
+  s.add_dependency 'rails', '~> 5.1.0'
 
-  s.add_dependency 'rails-observers'
-
-  s.add_dependency 'active_model_serializers', '~> 0.8.2'
+  s.add_dependency 'active_model_serializers'#, '~> 0.8.2'
```
Comment (pvawser01) [inline koi.gemspec]: Should we be restricting the version range for these dependencies? Would need to determine what the version being used is and add that.

File: lib/koi.rb [hunk 5]
```diff
@@ -1,26 +1,21 @@
 require 'compass-rails'
 require 'sass-rails'
 require 'csv'
-require 'devise'
```
Comment (pvawser01) [inline lib/koi.rb]: Don't we need this?

File: Gemfile (hunk 1 above)
Comment (Obversity) [inline Gemfile]: Actually, I think that's been merged now. Good catch. I'll update this.

**Comment by pvawser01** [comment]
@Obversity This looks fine for a first pass upgrade. I'll not some things we will need to revisit.
I'll try running this locally now

File: koi.gemspec (hunk 4 above)
Comment (Obversity) [inline koi.gemspec]: Ah, yup, that is something I meant to do before merging. Good thinking! I'll do that now.

File: lib/koi.rb (hunk 5 above)
Comment (Obversity) [inline lib/koi.rb]: I moved it to the bottom of the file and commented it out, so it was grouped with the other gems that need re-enabling here once they've been updated and pushed.

File: db/migrate/20111228035029_create_nav_items.rb (hunk 3 above)
Comment (Obversity) [inline db/migrate/20111228035029_create_nav_items.rb]: Yeah, I didn't give too much thought about what should happen with these things. As we discussed, technically they should probably be `ActiveRecord::Migration[4.2]`, but they all seem to run just fine with `ActiveRecord::Migration[5.0]`, so I think we'll leave them. 

Any future migrations will generate with `[5.1]`.

File: README.md (hunk 2 above)
Comment (Obversity) [inline README.md]: Yup, will!

**Pull request merged**